himalaya/himalaya.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package himalaya
+
+import (
+	cmdfile "github.com/ethersphere/bee-repair/file"
+	"github.com/ethersphere/bee-repair/himalaya/internal/repair"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultMimeType     = "application/octet-stream"
+	limitMetadataLength = swarm.ChunkSize
+)
+
+var (
+	host      string // flag variable, http api host
+	port      int    // flag variable, http api port
+	ssl       bool   // flag variable, uses https for api if set
+	verbosity string // flag variable, debug level
+	encrypted bool   // flag variable, uses encryption
```

Comment:
So I have added pinning. I added the tests also. But after adding the tests I have realized that repair will just repair change manifest entries. The actual file reference will remain the same. So if we give 'pin' option only the new manifest chunks will be pinned.
Cases:
1. File was already pinned: This would work fine as new manifest will be pinned and the file chunks are already pinned. But in this case, we wont 'unpin' the chunks that previously represented the file.
2. File was not pinned before: In this case, we would just pin the manifest chunks. File reference will have to be pinned separately.

I would like to avoid having dependency on pinning service here. Else we might need to rewrite the implementation a bit.